<h1 style="text-align: center; color: #007acc; background-color: #f0f8ff; padding: 20px; border-radius: 10px; border-left: 5px solid #007acc;">🍽️ MiseEnPlace | Data Collection & Quality Diagnosis</h1>

## Overview
This notebook covers the first stage of the MiseEnPlace pipeline:
raw data ingestion and initial structural inspection.

**<span style="color: #28a745; font-weight: bold;">Dataset:</span>** Michelin Guide Restaurants  
**<span style="color: #007bff; font-weight: bold;">Source:</span>** Kaggle — ngshiheng/michelin-guide-restaurants-2021  
**<span style="color: #dc3545; font-weight: bold;">Goal:</span>** Understand the structure and quality of the raw data
before any transformation or hypothesis formation.

---

In [1]:
# Import dataset
import pandas as pd

df=pd.read_csv('../data/raw/michelin_my_maps.csv')
df.head()

,Name,Address,Location,Price,Cuisine,Longitude,Latitude,PhoneNumber,Url,WebsiteUrl,Award,GreenStar,FacilitiesAndServices,Description
0,ES:SENZ,"Mietenkamer Straße 65, Grassau, 83224, Germany","Grassau, Germany",€€€€,"Creative, Modern Cuisine",12.465618,47.785630,4.986414e+11,https://guide.michelin.com/en/bayern/grassau/r...,https://www.das-achental.com/essenz,3 Stars,0,"Air conditioning,Car park,Interesting wine list","From the get-go, the stage is set in grand sty..."
1,Tohru in der Schreiberei,"Burgstraße 5, Munich, 80331, Germany","Munich, Germany",€€€€,"Modern Cuisine, Japanese Contemporary",11.577475,48.137597,4.989215e+11,https://guide.michelin.com/en/bayern/mnchen/re...,https://schreiberei-muc.de/,3 Stars,0,"Interesting wine list,Notable sake list",It is absolutely worth climbing the 23 steps o...
2,Schwarzwaldstube,"Tonbachstraße 237, Baiersbronn, 72270, Germany","Baiersbronn, Germany",€€€€,"Classic French, Creative",8.358280,48.536911,4.974425e+11,https://guide.michelin.com/en/baden-wurttember...,https://www.traube-tonbach.de/kulinarik/schwar...,3 Stars,0,"Air conditioning,Car park,Great view,Interesti...","Schwarzwaldstube, the flagship restaurant of t..."
3,Victor's Fine Dining by christian bau,"Schlossstraße 27, Perl, 66706, Germany","Perl, Germany",€€€€,"Creative, Contemporary",6.387211,49.535173,4.968668e+10,https://guide.michelin.com/en/saarland/perl/re...,https://www.victors-fine-dining.de/,3 Stars,0,"Air conditioning,Car park,Interesting wine lis...",The way in which Christian Bau combines influe...
4,schanz. restaurant.,"Bahnhofstraße 8a, Piesport, 54498, Germany","Piesport, Germany",€€€€,"Modern French, Creative",6.926600,49.877600,4.965079e+10,https://guide.michelin.com/en/rheinland-pfalz/...,https://www.schanz-restaurant.de/de/,3 Stars,0,"Air conditioning,Car park,Wheelchair access",It is thanks to Thomas Schanz that this small ...


## Initial Dataset Assessment

After the first inspection, the 14 columns were divided into 3 groups:

### <span style="color: #dc3545;">🗑️ Irrelevant Columns</span>
`PhoneNumber` | `Url` | `WebsiteUrl` | `Description`

No analytical value for investor-focused analysis.
Will be dropped in `02_cleaning_eda.ipynb`.

---

### <span style="color: #ffc107;">🔍 Columns Requiring Further Investigation</span>
`Address` | `Latitude` | `Longitude` | `GreenStar` | `Cuisine`

- **Address vs Lat/Lon:** Both reference location — need to decide
  which best supports map visualization and regional analysis.
- **GreenStar:** Sustainability award — need to verify if it
  differentiates highly awarded restaurants from common ones.
- **Cuisine:** High potential but structurally complex — values mix
  culinary styles (Creative, Contemporary) with specialties
  (French, Japanese) in inconsistent order.

---

### <span style="color: #28a745;">✅ Relevant Columns</span>
`Name` | `Location` | `Price` | `Award` | `FacilitiesAndServices`

<table style="border-collapse: collapse; width: 100%;">
<tr style="background-color: #3b3b3bff;">
<th style="border: 1px solid #3b3b3bff; padding: 8px;">Column</th>
<th style="border: 1px solid #3b3b3bff; padding: 8px;">Role in Analysis</th>
</tr>
<tr>
<td style="border: 1px solid #3b3b3bff; padding: 8px;"><code>Name</code></td>
<td style="border: 1px solid #3b3b3bff; padding: 8px;">Restaurant identification for maps and ranking tables</td>
</tr>
<tr style="background-color: #3b3b3bff;">
<td style="border: 1px solid #3b3b3bff; padding: 8px;"><code>Location</code></td>
<td style="border: 1px solid #3b3b3bff; padding: 8px;">Geographic distribution of top-tier restaurants</td>
</tr>
<tr>
<td style="border: 1px solid #3b3b3bff; padding: 8px;"><code>Price</code></td>
<td style="border: 1px solid #3b3b3bff; padding: 8px;">Relationship between price level and award category</td>
</tr>
<tr style="background-color: #3b3b3bff;">
<td style="border: 1px solid #3b3b3bff; padding: 8px;"><code>Award</code></td>
<td style="border: 1px solid #3b3b3bff; padding: 8px;"><strong>Core column</strong> — anchor for all comparative analyses</td>
</tr>
<tr>
<td style="border: 1px solid #3b3b3bff; padding: 8px;"><code>FacilitiesAndServices</code></td>
<td style="border: 1px solid #3b3b3bff; padding: 8px;">Infrastructure patterns in awarded restaurants</td>
</tr>
</table>

With these columns, investors can identify the structural and
geographic patterns of 3-Star Michelin restaurants and use them
as a benchmark for investment decisions.

<h2 style="color: #007acc;">2. Data Quality Diagnosis</h2>

Before any analysis, we need to understand the quality of the data.
The first check is identifying missing values — and deciding
whether they require treatment or carry analytical meaning.

In [2]:
df.isnull().sum()

Name                        0
Address                     0
Location                    0
Price                       0
Cuisine                     0
Longitude                   0
Latitude                    0
PhoneNumber               438
Url                         0
WebsiteUrl               2719
Award                       0
GreenStar                   0
FacilitiesAndServices    1122
Description                 0
dtype: int64

> **Findings:**
> - Only 3 columns contain null values
> - `PhoneNumber` and `WebsiteUrl` are irrelevant — will be dropped
> - `FacilitiesAndServices` (1,122 nulls) is relevant and requires
>   deeper investigation: are nulls distributed evenly across all
>   restaurants, or concentrated in specific award categories?
> 
> This question will be answered after we map the award categories.
> 
> ---

In [8]:
order_award=['3 Stars','2 Stars','1 Star','Bib Gourmand','Selected Restaurants']
category_award=df['Award'].value_counts().reindex(order_award)
category_award

Award
3 Stars                   154
2 Stars                   536
1 Star                   3145
Bib Gourmand             3566
Selected Restaurants    11507
Name: count, dtype: int64

> **Findings:**
> - The dataset is heavily imbalanced: 11,507 unawarded restaurants
>   versus only 154 with 3 Stars
> - This imbalance means absolute counts alone can be misleading —
>   proportional analysis will be essential throughout this project
> - With award categories mapped, we can now investigate
>   the FacilitiesAndServices null distribution by category
> 
> ---

## 4. Columns Under Investigation

Three columns require deeper analysis before we decide
whether to keep or drop them: `Latitude/Longitude vs Address`,
`GreenStar`, and `Cuisine`.

---

### 4.1 Location: Address vs Latitude/Longitude

Both columns reference restaurant location, but serve different purposes.
`Latitude` and `Longitude` offer coordinate precision — essential for
map visualization in Power BI and for AI-based regional pattern analysis.
`Address` is human-readable but harder to process programmatically.

**Decision:** Keep `Latitude` and `Longitude`. Drop `Address`.

---

### 4.2 GreenStar

GreenStar is a separate Michelin award recognizing restaurants committed
to sustainable practices. The question is: does having a GreenStar
correlate with higher Michelin awards — making it relevant for
investor analysis?

In [20]:
print("FREQUNECIA ABSOLUTA")
df['GreenStar'].value_counts()


FREQUNECIA ABSOLUTA


GreenStar
0    18223
1      685
Name: count, dtype: int64

In [21]:
print("FREQUENCIA RELATIVA")
(df['GreenStar'].value_counts(normalize=True)*100).round(2)

FREQUENCIA RELATIVA


GreenStar
0    96.38
1     3.62
Name: proportion, dtype: float64

Less than 5% of restaurants have a GreenStar.
At first glance this could suggest GreenStar belongs to awarded restaurants —
but that assumption needs verification. Let's check the distribution
across award categories.

In [18]:
green_star_award=pd.crosstab(df['Award'],df['GreenStar']).reindex(order_award)
green_star_award

GreenStar,0,1
Award,,
3 Stars,118,36
2 Stars,469,67
1 Star,2861,284
Bib Gourmand,3478,88
Selected Restaurants,11297,210


> **Finding:** No meaningful relationship between GreenStar and award level.
> The ratio of 0s to 1s is consistent across all categories —
> 3-Star restaurants actually have fewer GreenStars than Selected Restaurants.
> 
> **Decision:** `GreenStar` is irrelevant for this analysis. Will be dropped.
> 
> ---

In [22]:
print(df['Cuisine'].head(20).to_string())

0                  Creative, Modern Cuisine
1     Modern Cuisine, Japanese Contemporary
2                  Classic French, Creative
3                    Creative, Contemporary
4                   Modern French, Creative
5          Creative French, Classic Cuisine
6                   Creative, International
7                            Classic French
8                  Creative, Modern Cuisine
9          Classic French, Seasonal Cuisine
10                 Modern Cuisine, Creative
11                                 Creative
12                        Creative, Seafood
13                                 Creative
14                                 Creative
15                   Creative, Contemporary
16               Creative, Regional Cuisine
17                                   Indian
18                   Modern Cuisine, French
19                         Regional Cuisine


In [23]:
print("\n=== Rows with multiple cuisines ===")
multiple=df['Cuisine'].str.contains(',',na=False).sum()
print(f"{multiple} rows contain more than one cuisine ({(multiple/len(df)*100).round(1)}%)")


=== Rows with multiple cuisines ===
6523 rows contain more than one cuisine (34.5%)


> **Finding:** A significant portion of restaurants list multiple cuisines
> per row in inconsistent order
> (e.g. `Creative, Modern Cuisine` and `Modern Cuisine, Creative`
> represent the same combination).
> 
> Additionally, the column mixes culinary **styles** (Creative, Contemporary)
> with culinary **specialties** (French, Japanese, Italian)
> — making direct comparison unreliable and potentially misleading
> for investment decisions.
> 
> **Decision:** `Cuisine` will be dropped from the analysis.

In [25]:
df.columns

Index(['Name', 'Address', 'Location', 'Price', 'Cuisine', 'Longitude',
       'Latitude', 'PhoneNumber', 'Url', 'WebsiteUrl', 'Award', 'GreenStar',
       'FacilitiesAndServices', 'Description'],
      dtype='str')

## 5. Relevant Columns Assessment

Based on the previous investigation, columns are split into two groups:

**Dropped:** `Address` | `Cuisine` | `Url` | `WebsiteUrl` 
| `GreenStar` | `Description`

**Kept:** `Name` | `Location` | `Price` | `Longitude` 
| `Latitude` | `Award` | `FacilitiesAndServices`

In [33]:
relevant_cols=['Name','Location','Price','Longitude','Latitude','Award','FacilitiesAndServices']
df[relevant_cols]

,Name,Location,Price,Longitude,Latitude,Award,FacilitiesAndServices
0,ES:SENZ,"Grassau, Germany",€€€€,12.465618,47.785630,3 Stars,"Air conditioning,Car park,Interesting wine list"
1,Tohru in der Schreiberei,"Munich, Germany",€€€€,11.577475,48.137597,3 Stars,"Interesting wine list,Notable sake list"
2,Schwarzwaldstube,"Baiersbronn, Germany",€€€€,8.358280,48.536911,3 Stars,"Air conditioning,Car park,Great view,Interesti..."
3,Victor's Fine Dining by christian bau,"Perl, Germany",€€€€,6.387211,49.535173,3 Stars,"Air conditioning,Car park,Interesting wine lis..."
4,schanz. restaurant.,"Piesport, Germany",€€€€,6.926600,49.877600,3 Stars,"Air conditioning,Car park,Wheelchair access"
...,...,...,...,...,...,...,...
18903,DOC,"Folgosa, Portugal",€€€,-7.682176,41.151575,Selected Restaurants,"Air conditioning,Car park,Great view,Interesti..."
18904,Prosa,"Aveiro, Portugal",€€€,-8.652609,40.642545,Selected Restaurants,Air conditioning
18905,Gustificio,"Carmignano di Brenta, Italy",€€,11.699030,45.628600,Selected Restaurants,"Air conditioning,Wheelchair access"
18906,Inka,"Verona, Italy",€€,10.993344,45.441347,Selected Restaurants,Air conditioning


Among the kept columns, three will drive the core comparative analysis —
each examined against `Award` category to extract investor insights:

- `Location` — which countries and cities concentrate the most
  awarded restaurants?
- `Price` — is there a relationship between price level
  and award category?
- `FacilitiesAndServices` — what infrastructure patterns
  differentiate 3-Star restaurants from unawarded ones?

The findings below document what each column contains and
what treatment is required before analysis.

### 5.1 Location

In [35]:
#Location- vamos verificar a frequencia de restaurantes por local
df['Location'].value_counts().head(20)

Location
Tokyo, Japan                      532
Paris, France                     455
London, United Kingdom            370
Singapore                         292
Osaka, Japan                      251
Kyoto, Japan                      251
New York, NY, USA                 248
Hong Kong, Hong Kong SAR China    219
Bangkok, Thailand                 185
Seoul, South Korea                177
Taipei, Taiwan                    158
Madrid, Spain                     151
Shanghai, Chinese Mainland        137
Dubai                             116
Washington, DC, USA               109
Guangzhou, Chinese Mainland       105
San Francisco, CA, USA            104
Chicago, IL, USA                  103
São Paulo, Brazil                  99
Beijing, Chinese Mainland          98
Name: count, dtype: int64

> **Finding:** Some entries contain only a city name without a country
> (e.g. `Singapore`, `Dubai`) — no comma separator.
> 
> **Required treatment:** Split into two columns — `City` and `Country`.
> This enables geographic analysis by both city and country level.
> Handled in `02_cleaning_eda.ipynb`.
> 
> ---

In [36]:
#Price
df['Price'].value_counts()

Price
€€      4850
€€€     3531
€€€€    2248
$$      1363
$$$     1016
$$$$     940
¥¥¥      786
$        592
¥¥       476
££       427
£££      405
€        388
¥¥¥¥     266
££££     264
¥        247
฿฿       212
฿        134
₩         93
₫         82
฿฿฿       68
₺₺        63
฿฿฿฿      61
₩₩        56
₫₫        55
₺₺₺       54
₩₩₩₩      48
₩₩₩       35
₺₺₺₺      27
₺         23
₫₫₫       22
₫₫₫₫      19
﷼﷼﷼       17
﷼﷼﷼﷼      13
£         13
﷼﷼        10
﷼          4
Name: count, dtype: int64

> **Finding:** Price is expressed in local currency symbols
> (€, $, ¥, £) with 1 to 4 characters indicating price level
> — 1 being budget, 4 being premium.
> 
> **Required treatment:** Normalize all currencies to a unified
> numeric scale (1–4) based on character count.
> Handled in `02_cleaning_eda.ipynb`.
> 
> ---

In [37]:
#FacilitiesAndServices
df['FacilitiesAndServices'].value_counts()

FacilitiesAndServices
Air conditioning                                                                       1964
Air conditioning,Terrace                                                               1152
Air conditioning,Counter dining                                                        1085
Terrace                                                                                1039
Air conditioning,Wheelchair access                                                      748
                                                                                       ... 
Air conditioning,Car park,Counter dining,Great view,Valet parking,Wheelchair access       1
Brunch,Counter dining                                                                     1
Shoes must be removed                                                                     1
Car park,Great view,Shoes must be removed,Terrace,Valet parking                           1
Air conditioning,Brunch,Car park,Great view,Terrace,Wheelc

> **Finding:** Multiple facilities listed per row, separated by commas.
> Direct comparison across award categories requires individual treatment.
> 
> **Required treatment:** Split and explode into individual rows
> for frequency analysis by award category.
> Handled in `02_cleaning_eda.ipynb`.

## 6. Next Steps — 02_cleaning_eda.ipynb

This notebook has completed its purpose: understanding the raw data
structure and identifying what needs to be done.

The following tasks will be executed in the next stage:

**1. Drop irrelevant columns**
`Address` | `Cuisine` | `Url` | `WebsiteUrl` | `GreenStar` | `Description` | `PhoneNumber`

**2. Transform relevant columns**
- `Location` → split into `City` and `Country`
- `Price` → normalize currency symbols to numeric scale (1–4)
- `FacilitiesAndServices` → split into individual values
  for frequency analysis

**3. Analyze transformed columns against `Award` category**
- Geographic distribution of awarded restaurants
- Price level vs award category
- Facility patterns in 3-Star restaurants

**4. Export clean dataset**
`../data/processed/michelin_cleaned.csv`

---
*Next notebook: `02_cleaning_eda.ipynb`*